packages/core/helper-plugin/src/components/PageSizeURLQuery.tsx

```diff
@@ -4,27 +4,36 @@
  *
  */
 
-import React from 'react';
+import * as React from 'react';
 
 import { Flex, Option, SingleSelect, Typography } from '@strapi/design-system';
-import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
 
-import { useTracking } from '../features/Tracking';
+import { useTracking, TrackingEvent } from '../features/Tracking';
 import { useQueryParams } from '../hooks/useQueryParams';
 
-export const PageSizeURLQuery = ({ trackedEvent, options, defaultValue }) => {
+export interface PageSizeURLQueryProps {
+  trackedEvent?: Extract<TrackingEvent, { properties?: never }>['name'];
+  options?: string[];
+  defaultValue?: string;
+}
```

Comment:
```suggestion
export interface PageSizeURLQueryProps<TOptions extends string[]> {
  trackedEvent?: Extract<TrackingEvent, { properties?: never }>['name'];
  options?: TOptions;
  defaultValue?: TOptions[number];
}
```

I think this will enforce that `defaultValue` is an item in `options`... I think